sunpy/timeseries/sources/goes.py

```diff
@@ -228,6 +229,8 @@ def _parse_netcdf(filepath):
         """
         with h5netcdf.File(filepath, mode="r", **XRSTimeSeries._netcdf_read_kw) as h5nc:
             header = MetaDict(OrderedDict(h5nc.attrs))
+            if len(header["id"].strip()) == 0:
+                header.update({"id": Path(filepath).name})
```

Comment:
In `map` we take the approach of not modifying any information in the header, but instead overriding properties. I think it would be good to follow that pattern in `timeseries`, if that's possible in this case (ie. with a custom `.observatory` property)?

Reply:
yeah thats a good point. I was only modifying it here if it was empty as the filename is in the 'id' meta for the GOES-16 and 17 satellites, but not for 13, 14, 15 for some reason. 

I'll change this to another name